### PR TITLE
Pull and preload restorable apps

### PR DESCRIPTION
**Pull request by mike-sul**

### CODE REVIEW

**Comment by mike-sul** [comment]
Corresponding changes in repoz https://git.foundries.io/development/cloudplatforms/infrastructure/repoz/-/merge_requests/42

File: apps/pull.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    except Exception as exc:
+        logging.error('Failed to pull Target apps and images: {}\n{}'.format(exc, traceback.format_exc()))
+        exit_code = os.EX_SOFTWARE
+    return exit_code
+
+
+if __name__ == '__main__':
+    logging.basicConfig(format='%(asctime)s %(levelname)s: Apps Puller: %(module)s: %(message)s', level=logging.INFO)
+    args = get_args()
+    exit(main(args))
```
Comment (doanac) [inline apps/pull.py:50]: I believe python recommends `sys.exit` and `exit` is just targetted for the REPL.

File: apps/target_apps_fetcher.py [hunk 2]
```diff
@@ -76,3 +79,66 @@ def _fetch_apps(self, target, apps_shortlist=None, force=False):
...
+            with open(app_blob_file, 'wb') as f:
+                f.write(app_blob)
+
+            with tarfile.open(fileobj=BIO(app_blob)) as t:
+                t.extract('docker-compose.yml', app_dir)
+
+            fetched_apps.append(ComposeApps.App(app_name, app_dir, validate=True))
+        return fetched_apps
+
+    def fetch_apps_images(self, graphdriver='overlay2', force=False):
```
Comment (doanac) [inline apps/target_apps_fetcher.py:127]: not for this, but some later cleanup. Seems like `graphdriver` is probably a part of a (sub)class's implementation details and not an actual parameter that needs to get passed around, right?

Comment (mike-sul) [reply apps/target_apps_fetcher.py:127]: yes

**Review by doanac — APPROVED** [review]

**Pull request merged**